doc/source/reference/c-api.generalized-ufuncs.rst

```diff
@@ -139,22 +140,35 @@ Notes:
    Each dimension name typically corresponds to one level of looping in the
    elementary function's implementation.
 #. White spaces are ignored.
+#. An integer as a dimension name freezes that dimension to the value,
+#. The name can be suffixed with a question mark, this make the dimension a
+   core dimension only if it exists on the input or output, otherwise 1 is used.
```

Comment:
We should make clear that the 1 will not be magically inserted if the core dimensions do not match

"otherwise 1 is used for a missing core dimension" or so